internal/common/events/events.go

```diff
@@ -889,3 +883,1119 @@ func (e *ClusterDeregisteredAfterInactivityEvent) FormatMessage() string {
+        hostId,
+        infraEnvId,
+    )
+    eventsHandler.SendHostEventAtTime(ctx, ev, eventTime)
+}
+
+func (e *HostDeregisteredClusterEvent) GetName() string {
+    return "host_deregistered_cluster"
+}
+
+func (e *HostDeregisteredClusterEvent) GetSeverity() string {
+    return "info"
+}
+func (e *HostDeregisteredClusterEvent) GetClusterId() *strfmt.UUID {
+    return nil
+}
+func (e *HostDeregisteredClusterEvent) GetHostId() strfmt.UUID {
+    return e.HostId
+}
+func (e *HostDeregisteredClusterEvent) GetInfraEnvId() strfmt.UUID {
+    return e.InfraEnvId
+}
+
+func (e *HostDeregisteredClusterEvent) format(message *string) string {
+    r := strings.NewReplacer(
+        "{host_id}", fmt.Sprint(e.HostId),
+        "{infra_env_id}", fmt.Sprint(e.InfraEnvId),
+    )
+    return r.Replace(*message)
+}
+
+func (e *HostDeregisteredClusterEvent) FormatMessage() string {
+    s := "Host deregistered from cluster"
+    return e.format(&s)
+}
+
+//
+// Event host_installer_args_applied
+//
+type HostInstallerArgsAppliedEvent struct {
+    ClusterId strfmt.UUID
+    HostId strfmt.UUID
+    InfraEnvId strfmt.UUID
+    HostName string
+}
+
+var HostInstallerArgsAppliedEventName string = "host_installer_args_applied"
+
+func NewHostInstallerArgsAppliedEvent(
+    clusterId strfmt.UUID,
+    hostId strfmt.UUID,
+    infraEnvId strfmt.UUID,
+    hostName string,
+) *HostInstallerArgsAppliedEvent {
+    return &HostInstallerArgsAppliedEvent{
+        ClusterId: clusterId,
+        HostId: hostId,
+        InfraEnvId: infraEnvId,
+        HostName: hostName,
+    }
+}
+
+func SendHostInstallerArgsAppliedEvent(
+    ctx context.Context,
+    eventsHandler events.Sender,
+    clusterId strfmt.UUID,
+    hostId strfmt.UUID,
+    infraEnvId strfmt.UUID,
+    hostName string,) {
+    ev := NewHostInstallerArgsAppliedEvent(
+        clusterId,
+        hostId,
+        infraEnvId,
+        hostName,
+    )
+    eventsHandler.SendHostEvent(ctx, ev)
+}
+
+func SendHostInstallerArgsAppliedEventAtTime(
+    ctx context.Context,
+    eventsHandler events.Sender,
+    clusterId strfmt.UUID,
+    hostId strfmt.UUID,
+    infraEnvId strfmt.UUID,
+    hostName string,
+    eventTime time.Time) {
+    ev := NewHostInstallerArgsAppliedEvent(
+        clusterId,
+        hostId,
+        infraEnvId,
+        hostName,
+    )
+    eventsHandler.SendHostEventAtTime(ctx, ev, eventTime)
+}
+
+func (e *HostInstallerArgsAppliedEvent) GetName() string {
+    return "host_installer_args_applied"
+}
+
+func (e *HostInstallerArgsAppliedEvent) GetSeverity() string {
+    return "info"
+}
+func (e *HostInstallerArgsAppliedEvent) GetClusterId() *strfmt.UUID {
+    return &e.ClusterId
+}
+func (e *HostInstallerArgsAppliedEvent) GetHostId() strfmt.UUID {
+    return e.HostId
+}
+func (e *HostInstallerArgsAppliedEvent) GetInfraEnvId() strfmt.UUID {
+    return e.InfraEnvId
+}
+
+func (e *HostInstallerArgsAppliedEvent) format(message *string) string {
+    r := strings.NewReplacer(
```

Comment:
this comment is for a future PR, i think we can be more efficient with the replacements.
during the code generation, we can identify which place-holders should be replaced and include only them in this method instead all of other properties (even those that aren't part of the message).